modules/core/05-port/types/module.go

```diff
@@ -138,3 +138,9 @@ type Middleware interface {
 	IBCModule
 	ICS4Wrapper
 }
+
+// PacketDataUnmarshaler defines an optional interface which allows a middleware to
+// request the packet data to be unmarshaled by the base application.
+type PacketDataUnmarshaler interface {
```

Comment:
Is port the right place for this? Intuitively it feels like it belongs in channel or packet (but I may be wrong)

Reply:
We need to cleanup our api definitions. It's split between `modules/core/exported` and here. Most IBC module related interface functions are defined here, while packet/channel/connection/client interfaces are defined in exported. For now, I think it makes sense to leave this here since it should be implemented by an IBC module. Some day we will consolidate our API into a single location, hopefully in a way where changes to the IBC app interface doesn't cause versioning issues for light clients